packages/table/src/table2.tsx

```diff
@@ -231,36 +231,26 @@ export class Table2 extends AbstractComponent<Table2Props, TableState, TableSnap
     private refHandlers = {
         cellContainer: (ref: HTMLElement | null) => (this.cellContainerElement = ref),
         columnHeader: (ref: HTMLElement | null) => {
-            this.columnHeaderElement = ref;
             if (ref != null) {
-                this.columnHeaderHeight = Math.max(ref.clientHeight, Grid.MIN_COLUMN_HEADER_HEIGHT);
+                this.setState({ columnHeaderHeight: Math.max(ref.clientHeight, Grid.MIN_COLUMN_HEADER_HEIGHT) });
```

Comment:
Making this part of state so that `TableHotkeys` can access it and factor it into its calculations [here](https://github.com/palantir/blueprint/pull/6882/files#diff-8cd23c90b1959192d311f6112803042130c4c7709017e6cac6fd991ea89221beR316-R317).

Reply:
if this is just moved to state to share with `TableHotkeys` I'd prefer to instead have a new way of passing the existing ref storage, ex by passing `new TableHotkeys` a callback that returns `this.columnHeaderHeight`